app/src/androidTest/java/com/epfl/drawyourpath/mainpage/fragments/ChallengeFragmentTest.kt

```diff
@@ -0,0 +1,393 @@
+package com.epfl.drawyourpath.mainpage.fragments
+
+import android.os.Bundle
+import android.view.View
+import androidx.fragment.app.testing.FragmentScenario
+import androidx.recyclerview.widget.RecyclerView
+import androidx.test.espresso.Espresso.closeSoftKeyboard
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.UiController
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.action.ViewActions.replaceText
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.contrib.RecyclerViewActions
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.epfl.drawyourpath.R
+import com.epfl.drawyourpath.challenge.DailyGoal
+import com.epfl.drawyourpath.challenge.TemporaryUser
+import com.epfl.drawyourpath.challenge.Tournament
+import com.epfl.drawyourpath.challenge.Trophy
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.util.*
+
+
+@RunWith(AndroidJUnit4::class)
+class ChallengeFragmentTest {
+
+    private var userOnlyGoal =
+        TemporaryUser(LinkedList(mutableListOf(DailyGoal(5.0, 15.0, 1, 1.56, 12.0, 1))))
+
+
+    @Test
+    fun displayCorrectGoal() {
+        val bundle = Bundle()
+        bundle.putSerializable("user", userOnlyGoal)
+
+        val scenario = FragmentScenario.launchInContainer(
+            ChallengeFragment::class.java,
+            bundle,
+            R.style.Theme_Bootcamp
+        )
+
+        //check that distance is correct
+        onView(withId(R.id.goals_view))

```

Comment:
This three tests should be separated in three independent tests because they are related to different date. It will be useful when this will linked directly to the user account

Reply:
I separated them into different tests

Reply:
Thanks for your change, the rest result will be more easy to understand.